Sprint #1

denishamann opened the pull request.

Commit:
feat(scan): adding blueprints mvp, barcodescanner, retrofit, test with espresso, travis.yml

CharlieMcFly committed:
fix(api):  Modif server api & autorisation of camera

modification dans l'api du serveur
autorisation de la camera dans les tests
ajout de log pour une meilleure comprenhension du parcours de l'ean dans l'application.

CharlieMcFly committed:
feat(scan): Ajout d'une Progress Bar Material Design

ajout de la librairy dans le gradle
ajout de la progressbar dans l'activity et dans le fragment

CharlieMcFly committed:
fix(travis): mise a jout de la version de la support de 24.2.0 à 24.2.1

CharlieMcFly committed:
fix(travis): test avec espresso

ajout du test sur l'activité product
ajout d'une tache dans le build.gradle
ajout du test dans le .travis

Commit:
feat (scan): display d'un produit, changement du modèle de produit.

Display d'un produit beaucoup plus agréable pour l'utilisateur.
On a ajouté les classes "Offer" et "QuantitativeValue" qui sont utilisées dans produit.
La classe produit est beaucoup plus complète et a évolué pour matcher avec l'api.

CharlieMcFly committed:
chore(travis): Change to make build work

Commit:
feat(product): fix problems with the view, added butterknife

Fix problems with hidden elements on product details.
Added ButterKnife for easier binding

CharlieMcFly committed:
feat(detailproduct): Mise en place du detail du produit via une nouvelle activité

on a ajouté une nouvelle activité pour créer apercevoir le produit scanner en gradant la méthode mvp

Commit:
docs(readme): Complete readme

Added screenshots

CharlieMcFly committed:
Merge pull request #2 from Oupsla/feat/scan

Feat/scan

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/UseCaseThreadPoolScheduler.java:
```diff
@@ -0,0 +1,76 @@
+ * See also {@link Executors} for a list of factory methods to create common
+ * {@link java.util.concurrent.ExecutorService}s for different scenarios.
+ */
+public class UseCaseThreadPoolScheduler implements UseCaseScheduler {
```
To me, all the part about Pool / Thread management is overkill. We just have a button & a scan. What motivated you to put all of this ?
I'm open to discussion but I would suggest to burn all of this.

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/data/source/Injection.java:
```diff
@@ -0,0 +1,50 @@
+ * at compile time. This is useful for testing, since it allows us to use
+ * a fake instance of the class to isolate the dependencies and run a test hermetically.
+ */
+public class Injection {
```
Consider using [Dagger](http://square.github.io/dagger/) to have a clean / classic dependency injection in your application. Doing so you'll be able to have clean test mock injection.

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/data/source/api/ApiService.java:
```diff
@@ -0,0 +1,34 @@
+ **/
+public class ApiService {
+
+    public static final String BASE_URL = "https://btrshopcloud.herokuapp.com/";
```
Use a resource string instead ?

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/detailsproduct/DetailsProductFragment.java:
```diff
@@ -0,0 +1,194 @@
+    public void showProduct(final Product product) {
+        if (product != null){
+
+            if (product.getBrand() != null){
```
You should probably create a method like:

```
private show(TextView view, String value) {
    if (value != null) {
        view.setText(value);
        view.setVisibility(View.VISIBLE);
    }
}
```

And call

```
show(brand, product.getBrand());
show(..., ...);
```

To avoid writing 50 times the same kind of block.

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/detailsproduct/domain/model/Offer.java:
```diff
@@ -0,0 +1,103 @@
+ * Created by denis on 03/11/2016.
+ */
+
+public class Offer implements Serializable {
```
Add a bit of Javadoc on EVERY classes.
- I'm not 100% convinced by the relevance of this class (but I might be wrong)

denishamann replied:
This class has been created to match with the API and the schema "offer" that appears in the schema "product" https://schema.org/Offer https://schema.org/Product

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/detailsproduct/domain/model/QuantitativeValue.java:
```diff
@@ -0,0 +1,96 @@
+ * Created by denis on 03/11/2016.
+ */
+
+public class QuantitativeValue implements Serializable {
```
Do we really need this ?

denishamann replied:
Yes I think it is a good idea to make an object for this. It's easier to handle the products API and it's schema. Also, it's good if we need to add other attributes in the future like for example minValue, maxValue.

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/products/ProductsActivity.java:
```diff
@@ -0,0 +1,138 @@
+public class ProductsActivity extends AppCompatActivity {
+
+
+    private static final String CURRENT_FILTERING_KEY = "CURRENT_FILTERING_KEY";
```
Used by... ?

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/products/ProductsActivity.java:
```diff
@@ -0,0 +1,138 @@
+                mProductsPresenter.getProduct(result);
+            }
+            if (resultCode == Activity.RESULT_CANCELED) {
+                //Write your code if there's no result
```
If you are not planning to give a feedback don't let empty conditional block.

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/products/ProductsFragment.java:
```diff
@@ -0,0 +1,155 @@
+ * Created by charlie on 20/10/16.
+ */
+
+public class ProductsFragment extends Fragment implements ProductsContract.View {
```
Check code style on this class. Lot of unecessary / empty lines

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/products/ProductsFragment.java:
```diff
@@ -0,0 +1,155 @@
+                .positiveText("Ok")
+                .show();
+
+        /* new AlertDialog.Builder(getContext())
```
Dead code

CharlieMcFly replied:
DONE : https://github.com/Oupsla/BtrShop-android/commit/b70c89df148a8a41706c3e206585b5fa05513834

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/scanner/ScannerActivity.java:
```diff
@@ -0,0 +1,241 @@
+import fr.lille1.univ.android.architecture.btrshop.scanner.domain.dialog.MessageDialogFragment;
+import me.dm7.barcodescanner.zxing.ZXingScannerView;
+
+public class ScannerActivity extends AppCompatActivity implements MessageDialogFragment.MessageDialogListener,
```
That's a LOT of code for just a scan.
I think most / all of the values can be set by default (like auto focus ? of course), always back camera (i'm not going to scan with my front cam), flash probably OFF, et format.. well... all ?

Cleaning all of these you'll have a short / sexy class.

CharlieMcFly replied:
DONE : https://github.com/Oupsla/BtrShop-android/commit/81bd53d5c02ffc998ac7c4d07746b5afae25b13a

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/scanner/domain/dialog/CameraSelectorDialogFragment.java:
```diff
@@ -0,0 +1,90 @@
+import fr.lille1.univ.android.architecture.btrshop.R;
+
+
+public class CameraSelectorDialogFragment extends DialogFragment {
```
Might be changed once the other camera class will be lighter. Dunno.

CharlieMcFly replied:
DONE : https://github.com/Oupsla/BtrShop-android/commit/81bd53d5c02ffc998ac7c4d07746b5afae25b13a

duribreux commented on app/src/main/java/fr/lille1/univ/android/architecture/btrshop/scanner/domain/dialog/FormatSelectorDialogFragment.java:
```diff
@@ -0,0 +1,96 @@
+import fr.lille1.univ.android.architecture.btrshop.R;
+import me.dm7.barcodescanner.zxing.ZXingScannerView;
+
+public class FormatSelectorDialogFragment extends DialogFragment {
```
Not sure about this either. Just dont display those choices and select all by default ?

CharlieMcFly replied:
DONE : https://github.com/Oupsla/BtrShop-android/commit/81bd53d5c02ffc998ac7c4d07746b5afae25b13a

duribreux reviewed (changes requested):
Add tests, coverage etc. Still a lot of work to do imo

duribreux commented on app/build.gradle:
```diff
@@ -0,0 +1,141 @@
+    buildToolsVersion rootProject.ext.buildToolsVersion
+
+    defaultConfig {
+        applicationId "fr.lille1.univ.android.architecture.btrshop"
```
Can you change to package name to io.btrshop

CharlieMcFly replied:
DONE : https://github.com/Oupsla/BtrShop-android/commit/81bd53d5c02ffc998ac7c4d07746b5afae25b13a

CharlieMcFly committed:
refactor : Modification pour la pull request

change package name to io.btrshop
remove dialog for the scanneractivity

CharlieMcFly committed:
refactor: Modification for pull request\n\nchange package to io.btrshop\nremove all the dialog for scanner activity

denishamann committed:
refactor(product): new method to avoid writing the same kind of block

CharlieMcFly committed:
refactor (productFragment) : clean dead code

remove comment
remove variable

CharlieMcFly committed:
refactor (productFragment) : clean empty line

remove unevessary
remove empty line

CharlieMcFly committed:
refactor (productActivity) : clean empty line & dead code

remove unused var
remove dead code

denishamann committed:
refactor(product): now storing the api url in a resource string

I needed a little trick for doing this because I was outside a context or an activity. See http://stackoverflow.com/a/4391811

denishamann committed:
docs(product): Product - Offer - QuantitativeValue

CharlieMcFly committed:
Merge pull request #3 from Oupsla/refactor/products

Refactor/products

CharlieMcFly committed:
Merge branch 'master' of https://github.com/Oupsla/BtrShop-android into refactor/pullrequest

CharlieMcFly committed:
fix(product) : File location change due to conflict

denishamann committed:
Merge pull request #4 from Oupsla/refactor/pullrequest

Refactor/pullrequest

CharlieMcFly committed:
refactor(dagger): Modification de l'architecture pour faire fonctionner l'application avec dagger

denishamann committed:
Merge pull request #5 from Oupsla/refactor/dagger

refactor(dagger): Modification de l'architecture pour dagger
